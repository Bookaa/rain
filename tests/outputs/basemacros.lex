keyword 'import'
name 'array'
newline
keyword 'import'
name 'dict'
newline
keyword 'import'
name 'macros'
newline
keyword 'let'
name 'fudge'
symbol '='
symbol '@'
name 'array'
symbol '.'
name 'new'
symbol '('
string 'one'
symbol ','
string 'two'
symbol ','
string 'three'
symbol ')'
newline
keyword 'let'
name 'node'
symbol '='
symbol '@'
name 'dict'
symbol '.'
name 'new'
indent
name 'tag'
symbol '='
string 'node'
newline
name 'body'
symbol '='
string 'body'
newline
dedent
newline
keyword 'let'
name 'main'
symbol '='
keyword 'func'
symbol '('
symbol ')'
indent
symbol '@'
name 'macros'
symbol '.'
name 'printall'
symbol '('
name 'fudge'
symbol ','
name 'fudge'
symbol '['
int 0
symbol ']'
symbol ','
name 'fudge'
symbol '['
int 1
symbol ']'
symbol ','
name 'fudge'
symbol '['
int 2
symbol ']'
symbol ')'
newline
symbol '@'
name 'macros'
symbol '.'
name 'printall'
symbol '('
string '----'
symbol ','
name 'node'
symbol ','
name 'node'
symbol '.'
name 'tag'
symbol ','
name 'node'
symbol '.'
name 'body'
symbol ')'
newline
dedent
newline
EOF
