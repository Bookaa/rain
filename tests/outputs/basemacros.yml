!program;1
stmts:
- !import;1
  name: array
  rename: null
- !import;1
  name: dict
  rename: null
- !import;1
  name: macros
  rename: null
- !assn;2
  export: false
  let: true
  lhs: !name;1
    value: fudge
  rhs: !expr_block;1
    expr: !name;1
      value: :base.array:new:0
    stmts:
    - !assn;2
      export: false
      let: true
      lhs: !name;1
        value: :base.array:new:0
      rhs: !table;1
        parent: null
    - !assn;2
      export: false
      let: false
      lhs: !index;1
        lhs: !name;1
          value: :base.array:new:0
        rhs: !int;1
          value: 0
      rhs: !str;1
        value: one
    - !assn;2
      export: false
      let: false
      lhs: !index;1
        lhs: !name;1
          value: :base.array:new:0
        rhs: !int;1
          value: 1
      rhs: !str;1
        value: two
    - !assn;2
      export: false
      let: false
      lhs: !index;1
        lhs: !name;1
          value: :base.array:new:0
        rhs: !int;1
          value: 2
      rhs: !str;1
        value: three
- !assn;2
  export: false
  let: true
  lhs: !name;1
    value: node
  rhs: !expr_block;1
    expr: !name;1
      value: :base.dict:new:0
    stmts:
    - !assn;2
      export: false
      let: true
      lhs: !name;1
        value: :base.dict:new:0
      rhs: !table;1
        parent: null
    - !assn;2
      export: false
      let: false
      lhs: !index;1
        lhs: !name;1
          value: :base.dict:new:0
        rhs: !str;1
          value: tag
      rhs: !str;1
        value: node
    - !assn;2
      export: false
      let: false
      lhs: !index;1
        lhs: !name;1
          value: :base.dict:new:0
        rhs: !str;1
          value: body
      rhs: !str;1
        value: body
- !assn;2
  export: false
  let: true
  lhs: !name;1
    value: main
  rhs: !func;1
    body: !block;1
      stmts:
      - !block;1
        stmts:
        - !call;1
          args:
          - !index;1
            lhs: !name;1
              value: fudge
            rhs: !int;1
              value: 0
          catch: false
          func: !name;1
            value: print
        - !call;1
          args:
          - !index;1
            lhs: !name;1
              value: fudge
            rhs: !int;1
              value: 1
          catch: false
          func: !name;1
            value: print
        - !call;1
          args:
          - !index;1
            lhs: !name;1
              value: fudge
            rhs: !int;1
              value: 2
          catch: false
          func: !name;1
            value: print
      - !block;1
        stmts:
        - !call;1
          args:
          - !str;1
            value: '----'
          catch: false
          func: !name;1
            value: print
        - !call;1
          args:
          - !index;1
            lhs: !name;1
              value: node
            rhs: !str;1
              value: tag
          catch: false
          func: !name;1
            value: print
        - !call;1
          args:
          - !index;1
            lhs: !name;1
              value: node
            rhs: !str;1
              value: body
          catch: false
          func: !name;1
            value: print
    params: []
