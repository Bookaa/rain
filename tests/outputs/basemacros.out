one
two
three
----
node
body
